src/Microsoft.ML.KMeansClustering/KMeansModelParameters.cs

```diff
@@ -4,31 +4,31 @@
 
 using Float = System.Single;
 
-using System;
-using System.IO;
-using Microsoft.ML.Runtime.Numeric;
-using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.KMeansClustering;
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Runtime.Data;
-using Microsoft.ML.Trainers.KMeans;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
 using Microsoft.ML.Runtime.Model;
 using Microsoft.ML.Runtime.Model.Onnx;
-using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Numeric;
+using System;
+using System.IO;
 using System.Collections.Generic;
 
-[assembly: LoadableClass(typeof(KMeansPredictor), null, typeof(SignatureLoadModel),
-    "KMeans predictor", KMeansPredictor.LoaderSignature)]
+[assembly: LoadableClass(typeof(KMeansModelParameters), null, typeof(SignatureLoadModel),
+    "KMeans predictor", KMeansModelParameters.LoaderSignature)]
 
-namespace Microsoft.ML.Trainers.KMeans
+namespace Microsoft.ML.KMeansClustering
```

Comment:
>Microsoft.ML.KMeansClustering [](start = 10, length = 29)

i wonder if we want to put all predictors in something:

Microsoft.ML.ModelParameters.KMeansClustering

Reply:
If the argument is that the namespace is "Trainers" and therefore only ITrainer implementors should live in that namespace, then I'd argue that we should not have "Trainers" in the namespace name, if you think it could cause confusion.

It would be very inconvenient to have *predictors* in a different namespace from their associated *trainers* (except in generalizable cases like linear predictors that can be produced by many things).

---
In reply to: [238843887](https://github.com/dotnet/machinelearning/pull/1739#discussion_r238843887) [](ancestors = 238843887)

Reply:
Maybe `Microsoft.ML.KMeans` would be a better namespace, but let's not have one namespace every time we devise a new type that isn't a trainer, a predictor, or whatever. 😛

Reply:
So both the trainer and predictor would be in `Microsoft.ML.KMeansClustering`. That is what I gather from @Zruty0 's description in #1699 .

Reply:
On #1318 there was an explicit ask to put all the trainers inside Microsoft.ML.Trainers or appropriate namaspaces. 

So if i read both those issues, the ModelParameters also go in the same namespace as the respective trainer; for this case Microsoft.ML.Trainers.Kmeans @TomFinley that sounds good?

Reply:
That makes sense. I'll stick to the conventions in #1318 and revert the namespace to `Microsoft.ML.Trainers.KMeans`.